test/http-client-benchmarks/src/main/java/software/amazon/awssdk/benchmark/VirtualThreadBenchmark.java

```diff
@@ -0,0 +1,219 @@
+        benchmark.setup();
+
+        virtualThreadExecutor = createVirtualThreadExecutor();
+
+        putKeyPrefix = benchmarkName + "-";
+    }
+
+    private SdkHttpClient configure(SdkHttpClient.Builder<?> builder) {
+        AttributeMap config = AttributeMap.builder()
+            .put(SdkHttpConfigurationOption.MAX_CONNECTIONS, maxConnections)
+            .build();
+
+        return builder.buildWithDefaults(config);
+    }
+
+    private ExecutorService createVirtualThreadExecutor() {
+        try {
+            // Use reflection to call Executors.newVirtualThreadPerTaskExecutor()
+            Method method = Executors.class.getMethod("newVirtualThreadPerTaskExecutor");
+            return (ExecutorService) method.invoke(null);
+        } catch (NoSuchMethodException e) {
+            throw new UnsupportedOperationException(
+                "Virtual threads are not available in this Java version. " +
+                "This benchmark requires Java 21 or higher.", e);
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException("Failed to create virtual thread executor", e);
+        }
+    }
+
+    @Benchmark
+    public void getObject(Blackhole blackhole) {
+        safeExecute(() -> {
+            ResponseInputStream<GetObjectResponse> object = s3Client.getObject(
+                r -> r.bucket(benchmark.bucketName()).key(benchmark.objKey(objectSize)),
+                ResponseTransformer.toInputStream());
+            blackhole.consume(object.response());
+            IoUtils.drainInputStream(object);
+        });
+    }
+
+    @Benchmark
+    public void putObject(Blackhole blackhole) {
+        String jmhThreadName = Thread.currentThread().getName();
+        safeExecute(() -> {
+            PutObjectResponse response = s3Client.putObject(
+                r -> r.bucket(benchmark.bucketName()).key(putKeyPrefix + jmhThreadName),
+                benchmark.requestBody(objectSize));
+            blackhole.consume(response);
+        });
+    }
+
+    @TearDown(Level.Trial)
+    public void tearDown() {
+        if (virtualThreadExecutor != null) {
+            virtualThreadExecutor.shutdown();
+            try {
+                if (!virtualThreadExecutor.awaitTermination(30, TimeUnit.SECONDS)) {
+                    virtualThreadExecutor.shutdownNow();
+                }
+            } catch (InterruptedException e) {
+                virtualThreadExecutor.shutdownNow();
+            }
+        }
+
+        if (benchmark != null) {
+            benchmark.cleanup();
+        }
+
+        if (s3Client != null) {
+            s3Client.close();
+        }
+    }
+
+    private void safeExecute(Runnable runnable) {
+        try {
+            virtualThreadExecutor.submit(runnable).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException("Error during execution", e);
```

Comment:
Should we call Thread.currentThread().interrupt() before rethrowing?

Reply:
I don't think it's necessary. Once there's an interrupt, the benchmark should fail and exit, which is what we want.